include/internal/catch_approx.hpp

```diff
@@ -40,7 +44,19 @@ namespace Detail {
             approx.scale( m_scale );
             return approx;
         }
+#if defined(__cplusplus) && __cplusplus >= 201103L
+        template <typename T, typename = typename std::enable_if<std::is_constructible<T, double>::value>::type>
```

Comment:
This is enabled when `T` is constructible from `double`. But you actually do the inverse below. Should use `std::is_constructible<double, T>` here.